app/lib/actions/candlepin/import_pool_handler.rb

```diff
@@ -13,19 +13,23 @@ def handle(message)
 
         message_handler = ::Katello::Candlepin::MessageHandler.new(message)
 
-        case message_handler.subject
-        when /entitlement\.created/
-          message_handler.import_pool_by_reference_id
-          message_handler.create_pool_on_host
-        when /entitlement\.deleted/
-          message_handler.import_pool_by_reference_id
-          message_handler.remove_pool_from_host
-        when /pool\.created/
-          message_handler.import_pool_by_entity_id
-        when /pool\.deleted/
-          message_handler.import_pool_by_entity_id
-        when /compliance\.created/
-          reindex_consumer(message_handler)
+        if message_handler.subscription_facet.nil?
```

Comment:
message_handler.subscription_facet() call this method: 
https://github.com/Katello/katello/blob/master/app/services/katello/candlepin/message_handler.rb#L51
which calls this method:
https://github.com/Katello/katello/blob/master/app/services/katello/candlepin/message_handler.rb#L42-L48

and this method blows up if entity exists, but there is no 'consumer' attribute to it, i get:
```
Failed Candlepin Event: undefined method `[]' for nil:NilClass
/home/vagrant/git/katello/app/services/katello/candlepin/message_handler.rb:46:in `consumer_uuid'
/home/vagrant/git/katello/app/services/katello/candlepin/message_handler.rb:51:in `subscription_facet'
/home/vagrant/git/katello/app/lib/actions/candlepin/import_pool_handler.rb:16:in `handle'
```

Reply:
which i think points out that this logic doesn't make sense, as you wouldn't have a sub_facet for some of the events (pool.created, pool.deleted)

Reply:
ya agreed, i'll move the check down into things that involve hosts.

Reply:
@jlsherrill fix wasn't too bad, was just a matter of

```patch
       def create_pool_on_host
+        return if self.subscription_facet.nil?
         pool = self.pool_by_reference_id
         ::Katello::SubscriptionFacetPool.where(subscription_facet_id: self.subscription_facet.id,
                                                pool_id: pool.id).first_or_create
       end
 
       def remove_pool_from_host
+        return if self.subscription_facet.nil?
         pool = self.pool_by_reference_id
         ::Katello::SubscriptionFacetPool.where(subscription_facet_id: self.subscription_facet.id,
                                                pool_id: pool.id).destroy_all
```

Will add tests and update PR now